src/pages/policies/pagination.test.tsx

```diff
@@ -0,0 +1,40 @@
+import { paginate } from './pagination';
+
+describe('paginate()', () => {
+  it('inserts the "..." on the right side only for the first 4 pages', () => {
+    expect(paginate(95, { currentPage: 1 })).toEqual([1, 2, 3, 4, 5, '...', 9, 10]);
+    expect(paginate(95, { currentPage: 2 })).toEqual([1, 2, 3, 4, 5, '...', 9, 10]);
+    expect(paginate(95, { currentPage: 3 })).toEqual([1, 2, 3, 4, 5, '...', 9, 10]);
+    expect(paginate(95, { currentPage: 4 })).toEqual([1, 2, 3, 4, 5, '...', 9, 10]);
+  });
+
+  it('inserts the "..." on both sides when past the 4th page', () => {
+    expect(paginate(95, { currentPage: 5 })).toEqual([1, '...', 4, 5, 6, '...', 10]);
+    expect(paginate(95, { currentPage: 6 })).toEqual([1, '...', 5, 6, 7, '...', 10]);
+  });
+
+  it('inserts the "..." on the left side only for the last 4 pages', () => {
+    expect(paginate(95, { currentPage: 7 })).toEqual([1, 2, '...', 6, 7, 8, 9, 10]);
+    expect(paginate(95, { currentPage: 8 })).toEqual([1, 2, '...', 6, 7, 8, 9, 10]);
+    expect(paginate(95, { currentPage: 9 })).toEqual([1, 2, '...', 6, 7, 8, 9, 10]);
+    expect(paginate(95, { currentPage: 10 })).toEqual([1, 2, '...', 6, 7, 8, 9, 10]);
+  });
+
+  it('accepts a custom page size', () => {
+    expect(paginate(42, { pageSize: 5, currentPage: 4 })).toEqual([1, 2, 3, 4, 5, '...', 8, 9]);
+    expect(paginate(42, { pageSize: 5, currentPage: 5 })).toEqual([1, '...', 4, 5, 6, '...', 9]);
+    expect(paginate(42, { pageSize: 5, currentPage: 6 })).toEqual([1, 2, '...', 5, 6, 7, 8, 9]);
+  });
+
+  it('handles when the current page is out of bounds', () => {
+    expect(paginate(200, { currentPage: -1 })).toEqual([1, 2, 3, 4, 5, '...', 19, 20]);
```

Comment:
One thing that may be problematic is that if we want to support clicking on the `...` and doing some special behaviour. I don't think we care about it now, just giving a heads up.

Reply:
If we end up needing that, then it shouldn't require too much effort (the component could handle that)